extensions/wikia/WikiFactory/WikiFactory.php

```diff
@@ -1217,29 +1238,36 @@ static public function getLocalEnvURL( $url ) {
 		$devboxRegex = '/\.([^\.]+)\.wikia-dev\.com$/';
 
 		if ( preg_match( $devboxRegex, $server, $groups ) === 1 ) {
-			$devbox = $groups[1];
+			$devbox = $groups[ 1 ];
 		} else {
 			$devbox = '';
 		}
 
 		$server = str_replace( '.' . $devbox . '.wikia-dev.com', '', $server );
-		$server = str_replace( '.wikia.com', '', $server );
+		$server = str_replace( static::WIKIA_TOP_DOMAIN, '', $server );
+
+		// determine the environment we want to get url for
+		$environment = (
+			$forcedEnv &&
+			$forcedEnv !== WIKIA_ENV_DEV &&
+			$forcedEnv !== WIKIA_ENV_SANDBOX
+		) ? $forcedEnv : $wgWikiaEnvironment;
 
 		// put the address back into shape and return
-		switch($wgWikiaEnvironment) {
+		switch ( $environment ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a "case default" clause to this "switch" statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)